tests_and_analysis/test/script_tests/test_dispersion.py

```diff
@@ -3,6 +3,10 @@
 import pytest
 import numpy as np
 import numpy.testing as npt
+import matplotlib
+# Default mpl backend fails on system without $DISPLAY set for < 3.0.0
+if int(matplotlib.__version__.split('.')[0]) < 3:
+    matplotlib.use('Agg')
```

Comment:
See comment on comparing pint versions. Apply also to test_dos.py, test_sphere_sampling.py

An alternative would be to have the old libs tests run with xvfb so there is a virtual display in place; this is pretty common practice for testing.

I don't think it hurts to run some tests with the Agg backend, there's just a question of whether that should be 100% correlated with testing an old version of numpy :shrug:

Reply:
This ended up being surprisingly easy - I added the pytest-xvfb plugin and it all just worked!